### PR TITLE
Feature/set up claim backend

### PR DESCRIPTION
**Pull request by VargaElod23**
# Generic Pull Request template

## Linked issue

Please link the issue:

ex. Closes #<33>.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.
- [ ] The changes impact front-end.

## For changes impacting frontend

Please leave some screenshots/video recordings of the newly implemented interface and the Figma link for its initial design.

### Screenshots/ recordings

=> Multimedia here.

### Figma link

=> Figma link here.

## Description

Please provide a short description about what the PR contains.

## Funny gif

Leave a funny gif for faster review.

### CODE REVIEW

File: services/hype-backend/src/modules/reward/reward.controller.ts [hunk 1]
```diff
@@ -15,9 +31,27 @@ export class RewardController {
...
     description: 'Returns a newly inserted pool reward',
   })
+  @ApiUnauthorizedResponse({ description: 'You need a valid token' })
+  @ApiNotFoundResponse({ description: 'Claim not found' })
   public async accrueReward(
     @Body() rewardToAccrue: RewardDto,
   ): Promise<HypeReward> {
-    return await this.rewardService.accrueRewards(rewardToAccrue);
+    try {
+      return await this.rewardService.accrueRewards(rewardToAccrue);
```
Comment (VargaElod23) [inline services/hype-backend/src/modules/reward/reward.controller.ts:40]: Comment this endpoint for now

File: services/hype-backend/src/modules/reward/reward.service.ts [hunk 2]
```diff
@@ -35,4 +31,50 @@ export class RewardService {
     const saved = await this.rewardRepository.save(newReward);
     return saved;
   }
+
+  async releaseRewardHash(address: string) {
+    const rewardsOfAddress = await this.rewardRepository.findBy({
+      rewardee: Raw((alias) => `LOWER(${alias}) LIKE LOWER(:address)`, {
+        address,
```
Comment (VargaElod23) [inline services/hype-backend/src/modules/reward/reward.service.ts:38]: Take only the unclaimed.

File: services/hype-backend/src/modules/reward/reward.service.ts [hunk 3]
```diff
@@ -1,29 +1,49 @@
...
+    return await this.rewardRepository.find();
+  }
 
-  private getClaimContractInstance() {
-    const escrowContractInstance = this.blockchainService.getContractInstance(
-      ContractTypes.CLAIM,
-      this.ethereumConfig.claimContractAddress,
-      ProviderType.WALLET,
+  async getRewardSummaryForAddress(address: string): Promise<RewardStateDto> {
+    const rewardsOfAddress = await this.rewardRepository.findBy({
```
Comment (Malak67) [inline services/hype-backend/src/modules/reward/reward.service.ts:31]: Maybe add an exception below this. If (rewardsOfAddress?.length === 0) { throw new NotFoundException()} or something similar, before using the .filter on it.

Comment (Malak67) [reply services/hype-backend/src/modules/reward/reward.service.ts:31]: Saw you added this check on the next method, so you can add it here as well.

File: services/hype-backend/src/modules/reward/reward.controller.ts [hunk 4]
```diff
@@ -1,23 +1,79 @@
...
 @ApiTags('rewards')
 @Controller('rewards')
 export class RewardController {
   constructor(private readonly rewardService: RewardService) {}
 
+  @Get()
+  @ApiResponse({
+    status: HttpStatus.OK,
+    type: [RewardDto],
+    description: 'Returns a newly inserted pool reward',
```
Comment (Malak67) [inline services/hype-backend/src/modules/reward/reward.controller.ts:34]: description needs to be updated

File: services/hype-backend/src/modules/reward/reward.controller.ts [hunk 5]
```diff
@@ -1,23 +1,79 @@
...
+  })
+  async getAllRewardsForAddress(address: string): Promise<RewardStateDto> {
+    return await this.rewardService.getRewardSummaryForAddress(address);
+  }
+
   @Post()
```
Comment (Malak67) [inline services/hype-backend/src/modules/reward/reward.controller.ts:50]: I think we should have some authorization on these endpoints. For now you can add `@UseGuards(WalletGuard)` and afterwards we will see if we should add another custom guard.

**Review by Malak67 — CHANGES_REQUESTED** [review]
Small changes needed

**Review by Malak67 — APPROVED** [review]

**Pull request merged**